### PR TITLE
[FIX] constrain complement() functions

### PR DESCRIPTION
**Pull request by h-2**
Does someone want to look at this or can I merge it after CI passes?

### CODE REVIEW

File: include/seqan/modifier/modifier_shortcuts.h [hunk 1]
```diff
@@ -200,13 +200,13 @@ typedef ReverseComplementString<Rna5String>::Type    Rna5StringReverseComplement
  */
 
 template <typename TSequence >
-inline void complement(TSequence & sequence)
+inline SEQAN_FUNC_ENABLE_IF(IsSequence<TSequence>, void) complement(TSequence & sequence)
```
Comment (smehringer) [inline include/seqan/modifier/modifier_shortcuts.h]: Don't you want to fix the stringset overload below also?

Comment (h-2) [reply include/seqan/modifier/modifier_shortcuts.h]: The StringSet overload is already constrained to StringSets so it's not a problem!

Comment (smehringer) [reply include/seqan/modifier/modifier_shortcuts.h]: You sure that stringsets are physically restricted to contain sequences? :sweat_smile:

Comment (cpockrandt) [reply include/seqan/modifier/modifier_shortcuts.h]: But what about `reverseComplement(TSequence & sequence, Tag<TParallelTag> parallelTag)`? :stuck_out_tongue_closed_eyes:

Comment (h-2) [reply include/seqan/modifier/modifier_shortcuts.h]: What are you talking about? This is just about preventing the overload from being used for e.g. `int`.

**Pull request merged**